oml/functional/metrics.py

```diff
@@ -217,25 +239,34 @@ def calc_cmc(gt_tops: Sequence[BoolTensor], top_k: Tuple[int, ...]) -> List[Floa
         >>> gt_tops = [
         ...     BoolTensor([1, 0]),
         ...     BoolTensor([0, 1, 1]),
-        ...     BoolTensor([0, 0])
+        ...     BoolTensor([0, 0]),
+        ...     BoolTensor([])
         ... ]
-        >>> calc_cmc(gt_tops, top_k=(1, 2))
-        [tensor([1., 0., 0.]), tensor([1., 1., 0.])]
+        >>> n_gts = [2, 2, 1, 0]
+        >>> calc_cmc(gt_tops, n_gts, top_k=(1, 2))
+        [tensor([1., 0., 0., 1.]), tensor([1., 1., 0., 1.])]
     """
     check_if_nonempty_positive_integers(top_k, "top_k")
 
-    def cmc_single(is_correct: BoolTensor, k_: int) -> float:
-        value = float(is_correct[:k_].any())
-        return value
+    def cmc_single(is_correct: BoolTensor, n_gt: int, k_: int) -> float:
+        if n_gt == 0 and len(is_correct) == 0:
+            return 1.0
+        elif n_gt > 0 and len(is_correct) == 0:
+            return 0.0
+        else:
+            value = float(is_correct[:k_].any())
+            return value
 
     cmc = []
     for k in top_k:
-        cmc.append(FloatTensor([cmc_single(gts, k) for gts in gt_tops]))
+        cmc.append(
+            FloatTensor([cmc_single(gts, n_gt, k) for gts, n_gt in tqdm(zip(gt_tops, n_gts), desc="Computing CMC")])
```

Comment:
`desc=f"Computing CMC@{k}"`? 
"Computing CMC" better describes the outside loop.
The same comment for other functions.